# HG changeset patch
# Parent 8b4808ddd7484f0efa54ff7ee33d4b178e164366
# Parent  c8228ed01a36ea9e3c883bd03930bb453f766c04
Fix bitshift in test_utils_insert_mock_bulb for big endian systems

diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,8 +22,11 @@
 # TODO: we need at least 2.0.19-stable because of the logging defines
 FIND_PACKAGE(Event2 REQUIRED COMPONENTS core)
 FIND_PACKAGE(Endian REQUIRED)
+INCLUDE(TestBigEndian)
 INCLUDE(CompatTimeMonotonic)
 
+TEST_BIG_ENDIAN(LGTD_BIG_ENDIAN_SYSTEM)
+
 ### Global definitions #########################################################
 
 INCLUDE(AddAllSubdirectories)
@@ -32,6 +35,7 @@
 SET(CMAKE_C_FLAGS "-pipe -Wextra -Wall -Wstrict-prototypes -std=c99")
 
 ADD_DEFINITIONS("-DLGTD_SIZEOF_VOID_P=${CMAKE_SIZEOF_VOID_P}")
+ADD_DEFINITIONS("-DLGTD_BIG_ENDIAN_SYSTEM=${LGTD_BIG_ENDIAN_SYSTEM}")
 
 # Only relevant for the GNU libc:
 ADD_DEFINITIONS(
diff --git a/tests/core/tests_utils.c b/tests/core/tests_utils.c
--- a/tests/core/tests_utils.c
+++ b/tests/core/tests_utils.c
@@ -50,7 +50,10 @@
     union {
         uint8_t     as_array[LGTD_LIFX_ADDR_LENGTH];
         uint64_t    as_scalar;
-    } bulb_addr = { .as_scalar = htobe64(addr) >> 16 };
+    } bulb_addr = {
+        .as_scalar = LGTD_BIG_ENDIAN_SYSTEM ?
+            htobe64(addr) << 16 : htobe64(addr) >> 16
+    };
     struct lgtd_lifx_bulb *bulb = lgtd_lifx_bulb_open(gw, bulb_addr.as_array);
 
     SLIST_INSERT_HEAD(&gw->bulbs, bulb, link_by_gw);
